chef/site-cookbooks/wca/recipes/default.rb

```diff
@@ -167,6 +168,14 @@
 https = !node.chef_environment.start_with?("development")
 server_name = { "production" => "www.worldcubeassociation.org", "staging" => "staging.worldcubeassociation.org", "development" => "" }[node.chef_environment]
 
+# If /etc/ssh is not a symlink, back it up first.
+unless File.symlink?("/etc/ssh")
+  FileUtils.mv "/etc/ssh", "/etc/ssh-backup"
```

Comment:
What if `/etc/ssh-backup` already exists?

Reply:
I think that this command will fail? There are 2 cases in which I expect us to run chef:

1. **When bootstrapping a new server**: `/etc/ssh` is a directory, and `/etc/ssh-backup` does not exist.
2. **Updating an already bootstrapped server**: `/etc/ssh` is a symlink, and `/etc/ssh-backup` does not exist.

In 1), it's save to do the mv, because `/etc/ssh-backup` does not exist. In 2), we won't try to do the mv, because `/etc/ssh` is a symlink, and we only do the mv when it is not a symlink.

Let me know if I'm missing something obvious!

Reply:
Just asking. Good answer!

I didn't fully understand the scheme, and my "what happens when you do that a second time?" PTSD chimed in.